erpnext/hr/doctype/salary_structure/salary_structure.js

```diff
@@ -32,14 +32,120 @@ frappe.ui.form.on('Salary Structure', {
 	},
 	
 	refresh: function(frm) {
-		frm.trigger("toggle_fields")
+		frm.trigger("toggle_fields");
 		frm.fields_dict['earnings'].grid.set_column_disp("default_amount", false);
 		frm.fields_dict['deductions'].grid.set_column_disp("default_amount", false);
 		
 		frm.add_custom_button(__("Preview Salary Slip"),
 			function() { frm.trigger('preview_salary_slip'); }, "icon-sitemap", "btn-default");
+
+		frm.add_custom_button(__("Add Employees"),function () {
+			frm.trigger('add_employees')
+		})
 		
-	},	
+	},
+
+	add_employees:function (frm) {
+		var emp = new frappe.ui.Dialog({
+			title: __("Add Employees"),
+			fields: [
+				{fieldname:'company', fieldtype:'Link', options: 'Company', label: __('Company')},
+				{fieldname:'branch', fieldtype:'Link', options: 'Branch', label: __('Branch')},
+				{fieldname:'department', fieldtype:'Link', options: 'Department', label: __('Department')},
+				{fieldname:'designation', fieldtype:'Link', options: 'Designation', label: __('Designation')},
+				{fieldname:'base_variable', fieldtype:'Section Break'},
+				{fieldname:'base', fieldtype:'Currency', label: __('Base')},
+				{fieldname:'base_col_br', fieldtype:'Column Break'},
+				{fieldname:'variable', fieldtype:'Currency', label: __('Variable')},
+				{fieldname:'check_sec', fieldtype:'Section Break'},
+				{fieldname:'check_all', fieldtype:'Button', label: __('Check All')},
+				{fieldname:'check_col_br', fieldtype:'Column Break'},
+				{fieldname:'uncheck_all', fieldtype:'Button', label: __('Uncheck All')},
+				{fieldname:'check_col_br2', fieldtype:'Column Break'},
+				{fieldname:'set_employee', fieldtype:'Button', label: __('Set')},
+				{fieldname:'employees_sec', fieldtype:'Section Break', label: __('Employees')},
+				{fieldname:'employees_html', fieldtype:'HTML', label: __('Employees')}
+			]
+		});
+		frm.$empDialog = emp;
+		frm.trigger('getEmployees');
+		frm.$empDialog.get_input('company').on('change', function () {
+			frm.trigger('getEmployees');
+		});
+		frm.$empDialog.get_input('department').on('change', function () {
+			frm.trigger('getEmployees');
+		});
+		frm.$empDialog.get_input('branch').on('change', function () {
+			frm.trigger('getEmployees');
+		});
+		frm.$empDialog.get_input('designation').on('change', function () {
+			frm.trigger('getEmployees');
+		});
+		frm.$empDialog.get_input('check_all').on('click', function () {
+			frm.$empDialog.__check_all = true;
+			frm.trigger('employeeCheckToggle');
+		});
+		frm.$empDialog.get_input('uncheck_all').on('click', function () {
+			frm.$empDialog.__check_all = false;
+			frm.trigger('employeeCheckToggle');
+		});
+		frm.$empDialog.get_input('set_employee').addClass('btn-primary').on('click', function () {
+			frm.trigger('setEmployeesInChildTable');
+		});
+		frm.$empDialog.show();
+	},
+
+	getEmployees:function (frm) {
```

Comment:
We usually use `snake_case`, so for consistency sake, `get_employees` will be better :)